src/ModelFactory.php

```diff
@@ -23,7 +23,12 @@ public function __construct()
             $defaultAttributes = [];
         }
 
-        $factory = self::configuration()->factories()->create(static::class);
+        try {
+            $factory = self::isBooted() ? self::configuration()->factories()->create(static::class) : new static();
+        } catch (\ArgumentCountError $e) {
+            throw new \RuntimeException('Model Factories with dependencies (Model Factory services) cannot be created before foundry is booted.', 0, $e);
```

Comment:
I might even link to the docs here :p.

But also, one part of the solution is to make your args optional. But again, that might be best discussed by pointing to the docs. Heck, it might even make sense to have a FAQ/Errors section where you specifically guide users for common errors.

Reply:
#61